src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2687,6 +2687,49 @@ LogicalResult ONNXConstantOp::inferShapes(
 // Concat
 //===----------------------------------------------------------------------===//
 
+static LogicalResult verify(ONNXConcatOp op){
+  ONNXConcatOpAdaptor operandAdaptor =
+    ONNXConcatOpAdaptor(op);
+
+  int inputNum = op.getNumOperands();
+  // Check all inputs.
+  for (int i = 0; i < inputNum; ++i) {
+    if (!hasShapeAndRank(operandAdaptor.getOperands()[i])) {
+      // Won't be able to do any checking at this stage.
+      return success();
+    }
+  }
+  // Checking value of axis parameter.
+  auto commonType = operandAdaptor.getOperands()[0].getType().cast<RankedTensorType>();
```

Comment:
Is there a better way to get the axis parameter. Getting operand zero is not ideal because the reader does not know that op zero is the axis parameter.

Reply:
Hi Ettore,
could you explain a little bit more about this? I think the axis parameter is getting from attribute using `op.axis();`, and this `commonType` and `commonShape` are used to compare with other operands (make sure all operands have the same rank and shape).

Reply:
Mi main issue is with using `operandAdaptor.getOperands()[0]` in that line.  I would prefer `operandAdaptor.getOperands().first`

Reply:
@etiotto Please explain why the `first` is better. I am probably old school, I tend to use array notation. Always ready to learn new C++ conventions

Reply:
@AlexandreEichenberger actually the name of the member function is not `first()` but `front()`. The advantage of using `front()` rather than subscripting is that member function checks for an empty container:
```
 ReferenceT front() const {
    assert(!empty() && "expected non-empty range");
    return (*this)[0];
  }
```

Reply:
@etiotto are your requests satisfied? If so, can you approve? Tx